app/src/App/FileBrowser.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+namespace Dappur\App;
+
+/**
+ * Copied from https://github.com/vakata/jstree-php-demos/blob/master/filebrowser/index.php
+ * @SuppressWarnings(PHPMD.StaticAccess)
+ */
+
+// Limit To Log Folder
+ini_set('open_basedir', realpath(dirname(__FILE__) . DIRECTORY_SEPARATOR . ".." . DIRECTORY_SEPARATOR . ".." . DIRECTORY_SEPARATOR . "..".DIRECTORY_SEPARATOR."storage".DIRECTORY_SEPARATOR."log".DIRECTORY_SEPARATOR));
+
+class FileBrowser
+{
+    protected $base = null;
+
+    protected function real($path)
+    {
+        $temp = realpath($path);
+        if (!$temp) {
+            throw new Exception('Path does not exist: ' . $path);
+        }
+        if ($this->base && strlen($this->base)) {
+            if (strpos($temp, $this->base) !== 0) {
+                throw new Exception('Path is not inside base ('.$this->base.'): ' . $temp);
+            }
+        }
+        return $temp;
+    }
+    protected function path($id)
+    {
+        $id = str_replace('/', DIRECTORY_SEPARATOR, $id);
+        $id = trim($id, DIRECTORY_SEPARATOR);
+        $id = $this->real($this->base . DIRECTORY_SEPARATOR . $id);
+        return $id;
+    }
+    protected function id($path)
+    {
+        $path = $this->real($path);
+        $path = substr($path, strlen($this->base));
+        $path = str_replace(DIRECTORY_SEPARATOR, '/', $path);
+        $path = trim($path, '/');
+        return strlen($path) ? $path : '/';
+    }
+
+    public function __construct($base)
+    {
+        $this->base = $this->real($base);
+        if (!$this->base) {
+            throw new Exception('Base directory does not exist');
+        }
+    }
+    public function lst($id, $with_root = false)
```

Comment:
__[PHPMD]__

The method lst() has a Cyclomatic Complexity of 13. The configured cyclomatic complexity threshold is 10.


<!-- Comment by [Sider](https://sider.review) -->